include/uk/arch/ctx.h

```diff
@@ -56,12 +59,154 @@
 #define UKARCH_CTX_OFFSETOF_SP 8
 #endif
 
-#ifndef __ASSEMBLY__
+/* We must make sure that ECTX is aligned, so we make use of some padding,
+ * whose size is equal to what we need to add to UKARCH_ECTX_SIZE
+ * to make it aligned with UKARCH_ECTX_ALIGN
+ */
+#define UKARCH_EXECENV_PAD_SIZE					\
+	(ALIGN_UP(UKARCH_ECTX_SIZE,				\
```

Comment:
```C
/* Note: a has to be a power of 2 */
#ifndef ALIGN_UP
#define ALIGN_UP(v, a) (((v) + (a)-1) & ~((a)-1))
#endif

/* Note: a has to be a power of 2 */
#ifndef ALIGN_DOWN
#define ALIGN_DOWN(v, a) ((v) & ~((a)-1))
#endif
```

No reason to not work. The assembler resolves all of that macro mess to a constant. I can confirm it works as these are used for example in `plat/common/x86/syscall.S`

Reply:
Then this is resolved somewhere implicitly by the preprocessor. What I meant is that the essentials header is not included for assembly build because it is guarded at the top of syscall.h .